reputation/src/ars.rs

```diff
@@ -115,15 +113,16 @@ where
     where
         M: IntoIterator<Item = K>,
     {
-        if self.queue.len() >= self.capacity {
+        let identities: HashSet<K> = identities.into_iter().collect();
+
+        if !identities.is_empty() && self.queue.len() >= self.capacity {
```

Comment:
Why check `!identities.is_empty()`? If we want to assume that `identities` is never empty we should add `assert!(identities.is_empty())`.